src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcMaximum.scala

```diff
@@ -1,78 +0,0 @@
-/**
-  * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
-  *
-  * Licensed under the Apache License, Version 2.0 (the "License"). You may not
-  * use this file except in compliance with the License. A copy of the License
-  * is located at
-  *
-  *     http://aws.amazon.com/apache2.0/
-  *
-  * or in the "license" file accompanying this file. This file is distributed on
-  * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
-  * express or implied. See the License for the specific language governing
-  * permissions and limitations under the License.
-  *
-  */
-
-package com.amazon.deequ.analyzers.jdbc
-
-import java.sql.ResultSet
-
-import com.amazon.deequ.analyzers.Analyzers.{metricFromFailure, metricFromValue}
-import com.amazon.deequ.analyzers.MaxState
-import com.amazon.deequ.analyzers.jdbc.Preconditions._
-import com.amazon.deequ.analyzers.runners.EmptyStateException
-import com.amazon.deequ.metrics.{DoubleMetric, Entity}
-
-case class JdbcMaximum(column: String, where: Option[String] = None)
-  extends JdbcAnalyzer[MaxState, DoubleMetric] {
-
-  override def preconditions: Seq[Table => Unit] = {
-    hasTable() :: hasColumn(column) :: isNumeric(column) :: hasNoInjection(where) :: Nil
-  }
-
-  override def computeStateFrom(table: Table): Option[MaxState] = {
-
-    val connection = table.jdbcConnection
-
-    val query =
-      s"""
-         |SELECT
-         | MAX($column) AS col_max
-         |FROM
-         | ${table.name}
-         |WHERE
```

Comment:
introduce a function that extends a query with a predicate if a where string has been given, copy and pasting this TRUE=TRUE makes the code unnecessarily complex